python/tvm/contrib/graph_runtime.py

```diff
@@ -33,17 +33,34 @@ def create(graph_json_str, libmod, ctx):
             graph_json_str = graph_json_str._tvm_graph_json()
         except AttributeError:
             raise ValueError("Type %s is not supported" % type(graph_json_str))
-    device_type = ctx.device_type
-    device_id = ctx.device_id
-    if device_type >= rpc_base.RPC_SESS_MASK:
+    if isinstance(ctx, TVMContext):
+        ctx = [ctx]
+    elif not isinstance(ctx, (list, tuple)):
+        raise ValueError("ctx has to be the type of TVMContext or a list of "
+                         "TVMCTVMContext")
+    for i, cur_ctx in enumerate(ctx):
+        if not isinstance(cur_ctx, TVMContext):
+            raise ValueError("ctx has to be the type of TVMContext or a list "
+                             "of TVMContext")
+        if cur_ctx.device_type == tvm.cpu(0).device_type or \
+           cur_ctx.device_type >= rpc_base.RPC_SESS_MASK:
+            ctx[0], ctx[i] = ctx[i], ctx[0]
```

Comment:
We should remove the "smart handling of swapping cpu into the first one. It should just be fine if the user wants to make it so.

Reply:
otherwise user might get confused on the behavior

Reply:
@tqchen Removed. I thought CPU should always be used as fallback. But any way, we generate host module and use it as the coordinator.

Reply:
Not necessarily, the logic should work out of the box, as long as the user knows what to do. For example, a user can map ctx[0] to gpu and ctx[1] to cpu